Restore facility search results summary

mpelzsherman opened the pull request:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/7555

## Description
Restore the facility search results summary, but without the number of results, since this can be inaccurate.

## Testing done
Manual + Unit + e2e (Cypress!)

## Screenshots
![Find_VA_Locations___Veterans_Affairs](https://user-images.githubusercontent.com/80267/88596310-d3e27f80-d032-11ea-98a0-70b2cdfc0fce.png)

## Acceptance criteria
For all Facility Locator searches:
- [x] Results summary subhead displays when search results are returned. 
- [x] Results summary subhead does not prematurely load
- [x] When changing criteria after a previous search, the result summary is not updated until the new search is executed
- [x] Heading levels do not skip
- [x] Focus moves from the Search button to this heading, as indicated in ticket #7240
- [x] Heading includes user search criteria (**Search results** for <**VA facility type**> near **"location parameters"**)
- [x] Heading updates with new search criteria if new search is initiated
- [x] No heading is seen while a search is in progress
- [x] Only the standard error displays if no search results are found (no heading displays)
- [x] Veterans using screenreaders understand the context of the search results list
- [x] Responsive design: all elements are fully visible
- [x] Keyboard test confirms focus
- [x] New functionality is covered by E2E test

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

mpelzsherman committed:
Restore facility search results summary

https://github.com/department-of-veterans-affairs/va.gov-team/issues/7555

mpelzsherman commented:
Question for @jenstrickland or @1Copenut - how do we make the H2 element focusable, since it's not a link?

jenstrickland commented:
The h2 gets a `tabindex="-1"`. 

To see an example on VA.gov, go to https://www.va.gov/find-forms/ and enter a search query, then inspect the text with the yellow focus outline.

mpelzsherman committed:
Fix unit tests

mpelzsherman committed:
Add SearchResultsHeader component

mpelzsherman commented:
OK - I added tabindex="-1" to the h2, but that alone doesn't cause it to accept focus with the tab key. In fact, the /find-forms search results header isn't tab-focusable either. We have Javascript code that sets focus on the h2 after a search is executed. I've done the same for our facility locator form. It has the effect of causing screen readers to say "Results for ..." after the search is performed, which seems to be what we want. I'm not sure the `tabindex` is doing anything here, but I've added it anyway.

jenstrickland commented:
I'll tag @ncksllvn and @1Copenut to help with this. With my SOCKS issues, I am not as savvy — yet!

1Copenut commented:
@mpelzsherman The `tabindex="-1" allows the H2 to take focus via scripting. I checked the preview and it appears to be focusing correctly, if just a touch too early. Can you move focus to the H2 after data is returned (promise fulfilled?) so users will hear the announcement and be able to cycle through results immediately?

Non-native focusable elements must have a tab index of 0 or higher to receive TAB keyboard focus. We recommend not setting any tab index expect -1 for JavaScript, and 0 for inserting an element into the native tab order.

jenstrickland commented:
@1Copenut Thank you! I'd love to document this somewhere. Do you have an idea for where we might capture this? I initially thought the [Common Issues](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/accessibility/common-issues#common-accessibility-issues) area, but then couldn't locate another ticket.  Iirc, you've explained this to a number of developers and designers in the past 9 months. I'd love to be able to point people to a place. :)

mpelzsherman committed:
Set focus on results header when search is completed.

Also, hide results header while search is in progress.

mpelzsherman committed:
Added unit test for SearchResultsHeader

mpelzsherman commented on src/applications/facility-locator/components/FacilityTypeDescription.jsx:
```diff
@@ -16,20 +16,10 @@ const facilityName = (query, location) => {
+  <div className="vads-u-margin-bottom--1">
+    {from === 'FacilityDetail' && <strong>Facility type: </strong>}
+    {facilityName(query, location)}
+  </div>
```
Removing all-caps treatment of facility name per 508 guidelines.

mpelzsherman commented on src/applications/facility-locator/containers/VAMap.jsx:
```diff
@@ -693,6 +688,7 @@ class VAMap extends Component {
                 </FeatureGroup>
               )}
             </Map>
+            {otherToolsLink}
```
Move this link to the bottom of the page per design input.

brangi commented on src/applications/facility-locator/sass/facility-locator.scss:
```diff
@@ -80,6 +80,12 @@ $facility-locator-color-gray: #ccc;
     border-bottom: solid thin $color-gray-lightest;
   }
 
+  .search-result-title {
```
is it possible to use value from  https://design.va.gov/utilities/ instead?

mpelzsherman committed:
Added cypress tests

mpelzsherman committed:
Render empty div to avoid flicker when searching

mpelzsherman committed:
fix e2e test failures

mpelzsherman committed:
Final styling tweaks

mpelzsherman committed:
fix unit test
